Do some Kubectl optimizations suggested by the golangci linter

i02sopop opened the pull request:
**What type of PR is this?**
/kind cleanup
/sig cli

**What this PR does / why we need it**:
The tool golangci-lint gives a bunch of warnings. This PR solves the easier/less controversial ones, so the code is simpler and a little bit more optimal, since it removes some `if` conditions.

**Special notes for your reviewer**:
This is my first PR, so any advice is more than welcome :).

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

i02sopop commented:
/assign @soltysh

i02sopop commented:
/assign @smarterclayton

draveness commented on staging/repos_generated.bzl:
```diff
@@ -20,8 +20,8 @@ staging_repos = [
     "k8s.io/apiserver",
-    "k8s.io/cli-runtime",
     "k8s.io/client-go",
+    "k8s.io/cli-runtime",
```
What does the ci linter complain about this? Others LGTM.

i02sopop replied:
This one was suggested by the `hack/verify-bazel.sh` script, if I remember well, not by the linter. The linter only checks the `.go` files.

i02sopop replied:
The `verify` test have failed, and it says that I should run `./hack/update-bazel.sh`, but for me it generates the changes commited, should I roll back those changes? Or should I change the test?

draveness replied:
I think you should revert the change of this file.

draveness commented:
/ok-to-test
/priority backlog

draveness commented:
Please squash the commits into one.

Commit:
Do some Kubectl optimizations suggested by the golangci linter

The tool golangci-lint gives a bunch of warnings. This PR solves the easier/less controversial ones, so the code is simpler and a little bit more optimal, since it removes some if conditions.

i02sopop commented:
/test pull-kubernetes-node-e2e

dims commented:
/milestone v1.16

we are past the code freeze, so let's line this up for 1.16, please feel free to talk to the assignees when the 1.15 rush is over in a week or two.

i02sopop commented:
Any updates?

draveness commented:
> Any updates?

Please squash the commits into one, and ping the approver for reviews.

i02sopop commented:
@juanvallejo @deads2k Can you review the PR?

i02sopop commented:
/assign @smarterclayton

dims commented:
/assign @seans3

i02sopop commented:
@seans3 Any thoughts?

i02sopop commented:
@seans3 @juanvallejo @deads2k Can anyone review this PR?

ttousai commented:
Hello! I'm a bug triage shadow for the 1.16 release cycle and considering this issue is tagged for 1.16, I'd like to check its status. Code freeze is starting on August 29th (about 3 days from now), which means that this PR must merged by then.

Do we still target this issue to be fixed for 1.16?

i02sopop commented:
From my side yes, I think all the work is done for this issue, but it needs an approval, which I've been waiting three weeks since I've done the last changes to this PR.

i02sopop commented:
Can anyone review this PR? I don't know why it's taking so long to review, it's not so complicated.

i02sopop commented:
@juanvallejo @draveness @deads2k Can you please review this PR?

smourapina commented:
@i02sopop @juanvallejo @draveness @deads2k Bug triage for 1.17 here with a gentle reminder that code freeze for this release is on November 18. Is this PR still on track for 1.17?

i02sopop commented:
@smourapina I've been waiting for months to get this PR reviewed without any feedback, I don't know what else I need to do to get this PR reviewed :cry:

smourapina commented:
Pinging @soltysh @smarterclayton @seans3 as well, as they are listed as assignees: can you please assist in reviewing this PR for the 1.17 milestone?

i02sopop commented:
Sure

i02sopop commented:
Rebased with the last commits from master, for what is worth it.

i02sopop commented:
/test pull-kubernetes-e2e-gce-100-performance

soltysh reviewed (approved):
Good stuff! Thanks a ton!
/lgtm
/approve

soltysh commented:
/retest
